packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -202,35 +251,54 @@ export class LedgerKeyAgent extends KeyAgentBase {
   /**
    * @throws TransportError
    */
+  private static async createDeviceConnection(activeTransport: LedgerTransport): Promise<LedgerConnection> {
+    const deviceConnection = new LedgerConnection(activeTransport);
+    await this.testConnection(deviceConnection);
+    return deviceConnection;
+  }
+
+  /**
+   * @throws TransportError
+   */
+  // eslint-disable-next-line complexity
```

Comment:
This indicates that the function should *probably* be refactored

Reply:
Definitely. I added the comment in the early stage and overlooked it when finalising the implementation. I will refactor that method :+1: